### PR TITLE
Integrate JMTE hub to our existing infrastructure

### PR DESCRIPTION
**Pull request by yuvipanda**
- Use 'jupyter-meets-the-earth' rather than jmte as name, because the existing cluster is already called 'jmte'.
- Replicates config from https://github.com/2i2c-org/infrastructure/pull/436/files
  to the extent possible
- Uses our IRSA config for AWS permissions, rather than the eksctl created service account in use earlier.
- Uses CILogon+GitHub for authentication, rather than auth0+github
- Re-use the same EFS filesystem from before, avoiding the need to copy a few terabytes of data around
- Hub is now at jmte.2i2c.cloud, and the old URL (hub.jupyterearth.org) redirects here. Same for staging.
- Existing sqlite db has been copied over to the new cluster, so all existing users have access. The JupyterHub
  admin panel was used to manage access, so the hub was the source of user lists
- No prometheus existed on old cluster, so no prometheus data needs to be copied over.
- The staging hub config matches the prod hub config to the extent possible, matching rest of our infrastructure.
- SFTP service has been *removed* for now.
- Fix typo in our domain redirects code that prevented more than one redirect from being set

Ref https://github.com/2i2c-org/infrastructure/issues/2201

### CODE REVIEW

**Comment by yuvipanda** [comment]
I've deployed and set all this up already, as the existing cluster had gone down and I wanted to prevent disruptions.

**Review by consideRatio — APPROVED** [review]
Great work on this @yuvipanda!! LGTM!!!

**Comment by consideRatio** [comment]
Like for new hub setups, we want to provide permissions to work on this aws account for 2i2c engineers.

So, sending out invites is relevnt as well, but not directly related to this PR

File: config/clusters/jupyter-meets-the-earth/cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: jupyter-meets-the-earth
```
Comment (consideRatio) [inline config/clusters/jupyter-meets-the-earth/cluster.yaml:1]: Can you add a cloud console link like other cluster.yaml files has here?

File: config/clusters/jupyter-meets-the-earth/cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - name: staging
+    domain: staging.hub.jupytearth.org
+    helm_chart: daskhub
+    helm_chart_values_files:
+      - common.values.yaml
+      - staging.values.yaml
+      - enc-staging.secret.values.yaml
+  - name: prod
+    display_name: "Jupyter Meets the Earth"
+    domain: hub.jupytearth.org
```
Comment (consideRatio) [inline config/clusters/jupyter-meets-the-earth/cluster.yaml:22]: Hmmm should this be jmte.2i2c.cloud?

I got cert issues visiting hub.jupytearth.org

Comment (yuvipanda) [reply config/clusters/jupyter-meets-the-earth/cluster.yaml]: Ah, changed, but not sure why the redirects aren't working. Checking that now.

Comment (yuvipanda) [reply config/clusters/jupyter-meets-the-earth/cluster.yaml]: @consideRatio good catch, and fixed!

**Pull request merged**